parsec/core/cli/run.py

```diff
@@ -55,3 +57,121 @@ def run_mountpoint(config, device, mountpoint, timestamp, **kwargs):
         config = config.evolve(mountpoint_base_dir=Path(mountpoint))
     with cli_exception_handler(config.debug):
         trio_run(_run_mountpoint, config, device, timestamp)
+
+
+async def _import_file(workspace_fs, src, dest):
+    try:
+        await workspace_fs.touch(dest)
+    except FileExistsError:
+        await workspace_fs.truncate(dest, 0)
+    with open(src, "rb") as fd:
+        read_size = 0
+        while True:
+            chunk = fd.read(DEFAULT_BLOCK_SIZE)
+            if not chunk:
+                break
+            await workspace_fs.write_bytes(dest, chunk, read_size)
+            read_size += len(chunk)
+
+
+def _local_access_digests(workspace_fs, src):
+    digests = []
+    with open(src, "rb") as fd:
+        read_size = 0
+        while True:
+            chunk = fd.read(DEFAULT_BLOCK_SIZE)
+            if not chunk:
+                break
+            digests.append(HashDigest.from_data(chunk))
+            read_size += len(chunk)
+    return digests
+
+
+async def _update_file(workspace_fs, entry_id, path, absolute_path):
+    remote_file_manifest = await workspace_fs.remote_loader.load_manifest(entry_id)
+    local_access_digests = _local_access_digests(workspace_fs, path)
+    remote_access_digests = [access.digest for access in remote_file_manifest.blocks]
+    if local_access_digests != remote_access_digests:
+        await _import_file(workspace_fs, path, absolute_path)
+        print("update %s" % absolute_path)
+        await workspace_fs.sync()
+
+
+async def _create_path(workspace_fs, is_dir, path, absolute_path):
+    print("create %s" % absolute_path)
+    folder_manifest = None
+    if is_dir:
+        fs_path = FsPath(absolute_path)
+        await workspace_fs.mkdir(fs_path)
+        await workspace_fs.sync()
+        rep_info = await workspace_fs.path_info(fs_path)
+        folder_manifest = await workspace_fs.local_storage.get_manifest(rep_info["id"])
+    else:
+        await _import_file(workspace_fs, path, absolute_path)
+        await workspace_fs.sync()
+    return folder_manifest
+
+
+async def _clear_directory(parent_absolute_path, path, workspace_fs, folder_manifest):
+    local_children_keys = [c.parts[-1] for c in path.iterdir()]
+    for name, entry_id in folder_manifest.children.items():
+        if name not in local_children_keys:
+            absolute_path = f"{parent_absolute_path}{name}"
```

Comment:
Would be better to use type `Path` for `parent_absolute_path` (well I think we should use this everywhere it's possible !), hence we would avoid weird behavior if the `parent_absolute_path` lack a slash

```python
absolute_path = (parent_absolute_path / name).absolute()
```